client/components/NewEntryForm.js

```diff
@@ -16,27 +15,38 @@ class NewEntryForm extends Component {
     this.handleChange = this.handleChange.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
   }
+  async componentDidMount() {
```

Comment:
the best practice is to not have lifecycle hooks be async. you aren't awaiting anything here, anyway. the `getPrompt()` might be async, so you can let that do the thing that it needs to do.

in other words: in a lifecycle method, call an async function, but don't make the lifecycle method itself async.

Reply:
removed